interpreter/cling/lib/Interpreter/IncrementalJIT.cpp

```diff
@@ -360,7 +474,32 @@ IncrementalJIT::IncrementalJIT(
 
   // Create ObjectLinkingLayer with our own MemoryManager.
   Builder.setObjectLinkingLayerCreator([&](ExecutionSession& ES,
-                                           const Triple& TT) {
+                                           const Triple& TT)
+                                           -> std::unique_ptr<ObjectLayer> {
+    bool jitLink = false;
+    // Default to JITLink on macOS, as done in LLVM by
+    // LLJITBuilderState::prepareForConstruction.
+    if (TT.isOSBinFormatMachO() &&
+        (TT.getArch() == Triple::aarch64 || TT.getArch() == Triple::x86_64)) {
+      jitLink = true;
+    }
+    // Finally, honor the user's choice by setting an environment variable.
+    if (const char* clingJitLink = std::getenv("CLING_JITLINK")) {
+      jitLink = cling::utils::ConvertEnvValueToBool(clingJitLink);
+    }
```

Comment:
This commit should not be in here, we shouldn't add new "features" for patch releases